### PR TITLE
Add implementations documentation

### LINKED ISSUE #558: Docs Improvements
I wanted to list out some general docs improvements that would improve the site, for others to contribute, and also to keep track for myself. If anybody else has docs improvements that they would find valuable, feel free to pile on this issue and we can keep a running tracker.

- [ ] Make user guide Service names consistent (and consistent within concept docs too). There is foo-v1, foo-v2, foo, foo-canary, etc. Ideally, a small set of Deployment and Service resources should be able to be used to run every single one of the user guides so making the example names converge to only a few would allow this to happen.
- [ ] This needs discussion, but we should decide whether we want to specify an actual application Deployment within the Gateway docs so we can include end-to-end examples that contain real output from an app. I personally think this would be a good move because it would allow the user guides to have concrete steps in them rather than being theoretical examples as they are now. In a [previous version](https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/master/gateway#traffic-splitting) of the user guide I used [whereami](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/tree/master/whereami) but I'm happy to use any app that highlights the user guides we have.
- [ ] The [multi-ns user guide](https://kubernetes-sigs.github.io/gateway-api/multiple-ns/) should be reformatted into a similar style as the other guide
- [ ] The [TLS in HTTPRoute user guide](https://kubernetes-sigs.github.io/gateway-api/tls/) has some good conceptual content that is maybe better suited for [Gateway concepts](https://kubernetes-sigs.github.io/gateway-api/gateway/). Perhaps we should move some of that content there and slim down this TLS example to a smaller and self-contained guide.
- [ ] A TLSRoute user guide and also TLSRoute API Types page.
- [x] Addition of a "Gateway API implementations" page (or maybe add this content to [getting started](https://kubernetes-sigs.github.io/gateway-api/getting-started/) ) that lists out the current and planned implementations for Gateway. Ideally a user can go to an implementations page which shows how to install or enable that Gateway Controller and corresponding GatewayClasses in a K8s cluster. Then the user can deploy a reference Gateway from one of these GatewayClasses and continue with any of the standard user-guides, which should all theoretically function if that GatewayClass and controller is compliant.
- [ ] Gateway and Route ownership patterns need to be documented. There are several design patterns that make sense:
    - Single-tenant Gateways - Gateway and Route in same Namespace / Gateway and Route deployed by same owner
    - Shared Gateways - Gateways shared by Routes which are owned by different users (across Namespaces and also within the same Namespace)
    - Controlling hostname ownership - This is related to the route merging and conflicts, but a very common question is how to control which route-owners can own which hostnames. This can be done by specifying multiple listeners for the same port with different hostnames. Each of these listeners is bound to a different set of Routes, effectively giving those Routes exclusive ownership over the hostname. Since this is a design pattern, this capability is not really obvious to users and so it should be documented.
- [ ] We probably need some more conceptual content and examples for the [Gateway concepts](https://kubernetes-sigs.github.io/gateway-api/gateway/) page.
- [ ] Route merging and conflicts are only described in the API specification reference. This should have a section in the concept (or HTTPRoute) portion of the docs.

Happy to take edits or additions to this list. I just wanted to record what's on the top of my head before I forget :)

kind/docs

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

As per [k8s slack conversations][slack] and https://github.com/kubernetes-sigs/gateway-api/issues/558 this PR adds documentation to keep track of the downstream Gateway API implementations in other projects.

**Which issue(s) this PR fixes**:

Fixes #558 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:34]: TODO: need to verify that this is how the contour maintainers want to be represented

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:36]: I didn't see a `v0.1.0` implementation, but I could have missed it?

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:38]: We need to see if the Contour maintainers want to add more resources here, docs, blogs, e.t.c.

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:46]: @robscott is that how you and your team want to be represented? Please do feel free to make direct edits to get it the way you want it.

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:48]: @robscott I wasn't sure if you had any links to documentation, google docs, or blog posts you wanted to provide here yet?

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:54]: TODO: need to verify that this is how the Istio maintainers want to be represented

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:58]: Need to verify this, as I was just guessing when I made the PR draft, I'll do some more digging before I take this out of draft and make it ready for review.

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:80]: TODO: again need to make sure the Knative maintainers are on board with this representation. Also I'm _not sure_ whether they're actively working on an implementation or planning to. I'll do more digging before taking this out of draft state.

File: site-src/implementations.md
Comment (shaneutt) [inline site-src/implementations.md:92]: Similar to the above, need to check in with Traefik maintainers that they're happy with this representation.

**Review by shaneutt — COMMENTED** [review]
When I first made this doc it felt sort of odd to do write ups for other projects (I represent [Kong](https://github.com/kong)), but I tried to pull information from previous conversations in SIG networking zoom syncs and the K8s slack channel.

@robscott I would appreciate your review on this prior to moving it out of draft phase, perhaps I should stick with yours and mine implementations only for now? Not really sure since I'm new to the project, let me know :vulcan_salute:

File: site-src/implementations.md
Comment (robscott) [reply site-src/implementations.md:34]: @danehans @jpeach @youngnick Does this look good to you?

File: site-src/implementations.md
Comment (SantoDE) [inline site-src/implementations.md:92]: + TCP :)

Comment (SantoDE) [reply site-src/implementations.md:92]: Apart from that Traefik is good. Thanks a lot!

I was about to do the same this weekend 🥳

**Review by SantoDE — COMMENTED** [review]
See comments :)

File: site-src/implementations.md
Comment (robscott) [reply site-src/implementations.md:48]: Nothing public yet, but hope to have something out soon. I think this is good for now.

File: site-src/implementations.md
Comment (robscott) [reply site-src/implementations.md:46]: I think "Google Cloud" is the preferred term now. I'll let @mark-church weigh in on what we'd like to include here. I think we'll likely also want to add some GKE specific reference, but not quite sure how to integrate that.

File: site-src/implementations.md
Comment (howardjohn) [reply site-src/implementations.md:58]: we are importing the v0.2.x code now, but its in our `master` branch not a release yet and our docs aren't updated yet to point to the new "Gateway" name. Not sure what emoji that turns into :slightly_smiling_face:

File: site-src/implementations.md
Comment (robscott) [reply site-src/implementations.md:80]: @nak3 Do you have any insight you can provide here?

File: site-src/implementations.md
Comment (howardjohn) [inline site-src/implementations.md:82]: I think its a bit misleading? Knative is a user of the gateway API, not an implementation, afaik

File: site-src/implementations.md
Comment (stevesloka) [reply site-src/implementations.md:38]: We could also add some more info about the Contour Operator which helps to deploy things. @danehans

File: site-src/implementations.md
Comment (stevesloka) [reply site-src/implementations.md:36]: We started with v0.1.0 but never released it, so v0.2.0 is technically accurate. Also, I'm not sure that we're "fully implemented", it's very much a work in progress.

File: site-src/implementations.md
Comment (robscott) [inline site-src/implementations.md:23]: Although I like this table, our docs site rendering does not :(. Maybe we can just turn this into a list of implementations with approximate state in parentheses. Something like:

```
- [Contour][1] (alpha)
- [Google Cloud][2] (in progress)
...
```

File: site-src/implementations.md
Comment (SantoDE) [inline site-src/implementations.md:90]: We could add a hint about enabling and installing it with the helm chart. Not 100% sure this is the correct place tough :)

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:92]:
```suggestion
Traefik is currently working on implementing TCP, status updates and documentation will be provided here as the work progresses.
```

Comment (shaneutt) [reply site-src/implementations.md:92]: Just double checking this is what you wanted? Otherwise please drop a suggestion and I'll be happy to commit it :bow:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:82]: Roger that, I'll find someone from Knative and work on getting this written more appropriately for their liking :+1:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:36]: Ok thanks for the heads up! @robscott had suggested maybe dropping the implementation version stuff so perhaps I'll just go with that and then individual maintainers can put whatever they think are important notes about their implementations in their sections?

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:38]: Sounds good, _please_ do feel free to add github suggestions for what to change I'm happy to commit them I'm really seeing all of these sections as placeholders that eventually the maintainers will want to add their own writeup for once we have a loose format down anyhow :+1:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:46]:
```suggestion
The [Google Cloud][gcp] is a cloud computing platform and infrastructure provider.
```

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:48]: :+1:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:58]: Understood, thanks for the heads up! I'll try to get this updated to better reflect the state of things.

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:23]: Sure, I'll change it to that format instead, and perhaps we can wait until further down the road to see if people care to have a version implementation table, since it wouldn't seem to be all too important right now. :+1:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:90]: Sounds fine to me, is there just a link to that installation you want to list here?

File: site-src/implementations.md
Comment (mark-church) [reply site-src/implementations.md:46]: @shaneutt thanks for contributing this!

Here's a short statement we can use:

```
The [Google Kubernetes Engine (GKE)][gke] is a managed Kubernetes platform offered by Google Cloud. 

GKE is actively working towards an implementation of the Gateway API for orchestration of [Google Cloud Load Balancing][gclb]. The GKE Gateway Controller will become a managed component of GKE to automate and ease load balancing for containers.

[gke]:https://cloud.google.com/kubernetes-engine
[gclb]:https://cloud.google.com/load-balancingThe [Google Kubernetes Engine (GKE)][gke] is a managed Kubernetes platform offered by Google Cloud. 
```

File: site-src/implementations.md
Comment (mark-church) [reply site-src/implementations.md:54]: @howardjohn not sure if this should clarify that the current Gateway API is for Istio Ingress but not currently Istio sidecar mesh

File: site-src/implementations.md
Comment (nak3) [reply site-src/implementations.md:82]: Yeah, as @howardjohn also mentioned, Knative is an user of the gateway API but not an implementation. Knative is going to support the implemented Ingress like Istio, Kong, Contour etc.

File: site-src/implementations.md
Comment (nak3) [reply site-src/implementations.md:80]: Yes, It is a little bit weird that Knative is listed in the Implementation. Knative is going to support the Ingress which implemented gateway-api, so Knative is a kind of "user" rather than "implementation" as John Howard also mentioned below.

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:31]: I think we should just add more context in a follow up, if that's OK with you? Since this PR is mainly about adding the base page and getting some general information down. Let me know what you think, and feel free to unresolve and let me know your preference.

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md]: Sounds good, thank you! I've updated to match your contents: 267f6fbfea56821cbaeb1241c1842333abfb5ac9

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md]: I've updated this to be simpler and then more context can be added later in future iterations, lmkwyt :+1: 

5f51da790e097b5030548838e5bd51abceadc3fb

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md]: Ok, for now I've opted to remove it from the doc for now and then we can add further context as needed in other iterations: 7781c9acbc944c0c05270bc322bd2ba24509a17c

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md]: I've removed Knative for now as per above comments: 7781c9acbc944c0c05270bc322bd2ba24509a17c

We can come back around to documenting it further in a later iteration :+1:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:77]: For the time being I'm going to pass just so we can get the base page down, but we can definitely add further context as follow up :+1:

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:27]: Based on updates from @stevesloka below, I believe we should be all set now. Let me know if there's anything you still feel needs to be changed!

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:53]: Based on below responses I think we should be OK with Istio, but let me know if we're not!

**Comment by shaneutt** [comment]
At this point I think we've received enough maintainer feedback and updated for that feedback that this PR is ready to move from draft to ready. Please let me know any other considerations, or changes that you feel need to be made! :vulcan_salute:

**Comment by hbagdi** [comment]
Let's see if we can get an approval from maintainers from each implementation:

Contour: @stevesloka
GKE: @robscott @bowei
Istio: @howardjohn 
Kong: Approved by mysefl and @shaneutt 
Traefik: @SantoDE

**Review by howardjohn — APPROVED** [review]

File: site-src/implementations.md
Comment (bowei) [inline site-src/implementations.md:9]: can we make this a table?

```
| Implementation | Status |
| ---- | ---- | 
etc
```

Comment (bowei) [reply site-src/implementations.md:9]: Also, if we can put the version implemented (instead of just Alpha)
that would be useful for users

Comment (bowei) [reply site-src/implementations.md:9]: e.g. Supports `v1alpha1`

**Comment by bowei** [comment]
I put a hold -- let's collect ack from the impls

This looks great

File: site-src/implementations.md
Comment (shaneutt) [reply site-src/implementations.md:9]: When this was in draft it started out as a table, see:

https://github.com/kubernetes-sigs/gateway-api/pull/565#discussion_r583944465

However @robscott had particularly requested that I change it to the format you see here.

If Rob is OK with changing it back to a table but using the format you suggest above, that sounds good to me. Just want to check with him.

**Comment by robscott** [comment]
Just following up on Harry's summary from above, looks like we have most approvals, but still need a couple more. We currently have approvals for:
- [x] GKE
- [x] Istio
- [x] Kong

We still need approval from:
- [] Contour: @stevesloka
- [] Traefik: @SantoDE

/assign @stevesloka @SantoDE

File: site-src/implementations.md
Comment (bowei) [reply site-src/implementations.md:9]: This is a minor thing and we can change in a later diff. My thinking was that table is easier to deal with more info (say which versions are supported)

Comment (mark-church) [reply site-src/implementations.md:9]: I also think a table would be a bit cleaner (though we don't have to block this PR). The original table had the vendors as columns instead of rows which doesn't leave much room for written content. The following format would allow us to take the existing written content we have in the page and surface it through a table.

```
| Implementation | Status |
| [Project/Product](link) | Explanation of progress towards support and/or support levels (which versions) | 
| [Project/Product](link) | Explanation of progress towards support and/or support levels (which versions) | 
etc
```

I think support is a gray area right now since we are early in the spec so it's prob too early for a clean cut version/checkmark box.

File: site-src/implementations.md
Comment (mark-church) [inline site-src/implementations.md:10]: Should be:

```
- [Google Kubernetes Engine][2] (work in progress)
```

**Pull request merged**
